connect-inject/envoy_sidecar.go

```diff
@@ -55,13 +40,40 @@ func (h *Handler) envoySidecar(pod corev1.Pod) (corev1.Container, error) {
 			},
 		},
 		Command: cmd,
-		SecurityContext: &corev1.SecurityContext{
+	}
+
+	// If not running in transparent proxy mode and in an OpenShift environment,
+	// skip setting the security context and let OpenShift set it for us.
+	// When transparent proxy is enabled, then Envoy needs to run as our specific user
+	// so that traffic redirection will work.
```

Comment:
This comment makes more sense to me to be on line 54 right above `if tproxyEnabled || !h.EnableOpenShift`

Reply:
Good point, updated!